com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/context.py

```diff
@@ -158,82 +57,199 @@ def _print_process_stderr(process, fn):
-#      to the original toolkit layout. They are moved to the
-#      top level of the pypi package.
 #
-# SPL Toolkit (SPL focused):
-#   Streamsx Python packages are executed from opt/python/packages
-#   
-# This function determines the root of the SPL toolkit based
-# upon the existance of the '.toolkit' directory.
+# Submission of a python graph using the Java Application API
+# The JAA is reused to have a single set of code that creates
+# SPL, the toolkit, the bundle and submits it to the relevant
+# environment
 #
-def _get_toolkit_root():
-    # Directory of this file (streamsx/topology)
-    dir = os.path.dirname(os.path.abspath(__file__))
-
-    # This is streamsx
-    dir = os.path.dirname(dir)
-
-    # See if .toolkit exists, if so executing from
-    # a pip install
-    tk_root = os.path.join(dir, '.toolkit', 'com.ibm.streamsx.topology')
-    if os.path.isdir(tk_root):
-        return tk_root
-
-    # Else dir is tk/opt/python/packages/streamsx
-
-    dir = os.path.dirname(dir)
-    dir = os.path.dirname(dir)
-    dir = os.path.dirname(dir)
-    tk_root = os.path.dirname(dir)
-    return tk_root
-
-def _submitUsingJava(ctxtype, fn):
-    ctxtype_was = ctxtype
-    if ctxtype == "JUPYTER":
-        ctxtype = "STANDALONE"
-
-    tk_root = _get_toolkit_root()
-
-    cp = os.path.join(tk_root, "lib", "com.ibm.streamsx.topology.jar")
-
-    streams_install = os.environ.get('STREAMS_INSTALL')
-    if streams_install is None:
-        java_home = os.environ.get('JAVA_HOME')
-        if java_home is None:
-            raise "Please set the JAVA_HOME system variable"
-   
-        jvm = os.path.join(java_home, "bin", "java")
-        submit_class = "com.ibm.streamsx.topology.context.remote.RemoteContextSubmit"
-    else:
-        jvm = os.path.join(streams_install, "java", "jre", "bin", "java")
-        submit_class = "com.ibm.streamsx.topology.context.StreamsContextSubmit"
-        cp = cp + ':' +  os.path.join(streams_install, "lib", "com.ibm.streams.operator.samples.jar")
-
-    args = [ jvm, '-classpath', cp, submit_class, ctxtype, fn]
-    process = subprocess.Popen(args, stdin=None, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=0)
-    try:
-        stderr_thread = threading.Thread(target=_print_process_stderr, args=([process, fn]))
-        stderr_thread.daemon = True            
-        stderr_thread.start()
-        if not ctxtype_was == "JUPYTER":
-            stdout_thread = threading.Thread(target=_print_process_stdout, args=([process]))
-            stdout_thread.daemon = True
-            stdout_thread.start()                
-            process.wait()
-            return None
-        else:            
-            return process.stdout
-    except:
-        _print_exception("Error starting java subprocess for submission")
-        raise
+def submit(ctxtype, app_topology, config = None, username = None, password = None, rest_api_url = None, log_level=logging.INFO):
```

Comment:
Does changing the name of the argument break backwards compatibility (graph to app_topology) ?